submissions/DmytryjK/memory-pair-game/js/script.js

```diff
@@ -0,0 +1,204 @@
+      currentTime.innerText = localStorage.getItem('time');
+      bestResultsSheet.push(localStorage.getItem('time'));
+
+      if (bestResultsSheet.length > 1) {
+
+        bestTime.innerText = bestResultsSheet.reduce((prevItem, currentItem) => {
+          if (currentItem < prevItem) {
+            return currentItem;
+          } else {
+            return prevItem;
+          }
+        });
+
+      } else {
+        bestTime.innerText = localStorage.getItem('time');
+      }
+
+      localStorage.setItem('bestTime', bestTime.innerText);
+      clearInterval(timerInterval);
+
+      buttonToContinue.addEventListener('click', clickToContinue, false);
+    }
+    activeCards = [];
+  }
+}
+
+function clickToContinue(buttonToContinue) {
+  resetVariables(buttonToContinue);
+  createElementsOfCards();
+  currentListCards = document.querySelectorAll('.main__card');
+  resetTimer();
+  setTimeout(showAllCards, 400);
+  setTimeout(hideAllCards, 2000);
+}
+
+function hideActiveCards(activeCards) {
+  activeCards.forEach(activeCard => {
+    activeCard.classList.add('hide-card');
+  });
+
+  numbersOfCloseCards++;
+}
+
+function closeActiveCards(activeCards) {
+  activeCards.forEach(activeCard => {
+    activeCard.classList.remove('active-card');
+  });
+
+  activeCards = [];
+}
+
+function showAllCards() {
+  currentListCards.forEach(card => {
+    card.classList.add('active-card');
+  });
+}
+
+function hideAllCards() {
+  currentListCards.forEach(card => {
+    card.classList.remove('active-card');
+  });
+
+  clearInterval(timerInterval);
+  timerInterval = setInterval(timer, 1000);
+}
+
+function createElementsOfCards() {
+  doubleElementsOfCards = [...elementsOfCards, ...elementsOfCards];
+  doubleElementsOfCards.sort(() => Math.random() - 0.5);
+
+  for (let i = 0; i < (doubleElementsOfCards.length); i++) {
+    parrentCardBlock.innerHTML += `
+        <li class="main__card" data-name="${doubleElementsOfCards[i].hero}" data-numbers="${i}">
+            <img class="main__card-back" src="${srcBg}" alt="background_dota">
+            <img class="main__card-front" src="${doubleElementsOfCards[i].srcHero}" alt="${doubleElementsOfCards[i].hero}">
+        </li>
+        `;
+  }
+
+  mainCardsBlock.append(parrentCardBlock);
+  parrentCardBlock.addEventListener('click', onClickUserActions, false);
+}
+
+function resetTimer() {
+  seconds = 0;
+  minutes = 0;
+  currentTime.innerText = `0${minutes} m : 0${seconds} s`;
+}
+
+function timer() {
+  seconds++;
+
+  if (seconds >= 60) {
+    minutes++;
+    seconds -= 60;
+  }
+
+  if (seconds < 10 && minutes < 10) {
+    currentTime.innerText = `0${minutes} m : 0${seconds} s`;
+  } else if (seconds < 10) {
+    currentTime.innerText = `${minutes} m : 0${seconds} s`;
+  } else if (minutes < 10) {
+    currentTime.innerText = `0${minutes} m : ${seconds} s`;
+  } else {
+    currentTime.innerText = `${minutes} m : ${seconds} s`;
+  }
+}
+
+function resetVariables() {
```

Comment:
I'm not sure that the current function name describes all provided functionality: move removeEventListeners to another function or rename the current function.

Reply:
fixed